insta/src/snapshot.rs

```diff
@@ -517,39 +608,59 @@ impl Snapshot {
 }
 
 /// The contents of a Snapshot
-// Could be Cow, but I think limited savings
 #[derive(Debug, Clone)]
-pub struct SnapshotContents {
+pub enum SnapshotContents {
+    Text(TextSnapshotContents),
+
+    // This is in an `Rc` because we need to be able to clone this struct cheaply and the contents
+    // of the `Vec` could be rather large. The reason it's not an `Rc<[u8]>` is because creating one
+    // of those would require re-allocating because of the additional size needed for the reference
+    // count.
+    Binary(Rc<Vec<u8>>),
```

Comment:
We don't have to resolve before merging but — are we actually cloning this much? I tried removing the `Rc`, everything seemed to work, and I couldn't immediately see where we were cloning it.

Reply:
The place I saw the clone was in `runtime::record_snapshot_duplicate`. While that doesn't happen all the time I think it's worth it because wrapping it in an `Rc` is so easy.

Just as one reference point: the PDF files I'm snapshotting ended up being around 600kb usually. That's not a huge ammount of data, but compared to bumping a reference count it is a bunch of work and I think if we can prevent that copy so easily I would lean on the side of just doing it.

Reply:
Yes, good point. And seems difficult to change given it needs to hold the value throughout the test run...